app/services/player_ids_service.rb

```diff
@@ -0,0 +1,50 @@
+class PlayerIdsService
```

Comment:
I agree with @JuanfraM , all this code could be part of the NotificationService. In the initialize you can pass the specific user too and move the register method in there. If you do this, you need to add an if clause in the `notify` method to return when the matching_users list is empty.